docs/msbuild/updating-an-existing-application.md

```diff
@@ -79,7 +79,7 @@ Do not specify `ExcludeAssets=runtime` for the Microsoft.Build.Locator package.
 
 ### Register instance before calling MSBuild
 
-Add a call to the Locator API before calling any method that uses MSBuild.
+Add a call to the Locator API before calling any method that uses MSBuild. This is important because the just-in-time (JIT) compiler needs to understand the types of the objects it compiles, and when trying to build a project using a specific version of MSBuild, it has to inspect that version before it compiles the code in case those types have changed. Since the JIT compiler works at a method-level granularity, if the call to the Locator API is in the same method as the first use of an MSBuild type (even if it precedes it, and the code involving the type has not yet been executed), the JIT compiler will not be able to JIT the whole method, causing an error. Some types that require the call to the MSBuildLocator API to have already been called include ProjectRootElement, ProjectInstance, and BuildManager.
```

Comment:
I think this is overly detailed and would be better off with a good and bad example. Writing one up.

Reply:
Perhaps it's just because I think this sort of thing is really cool, but I would appreciate that level of detail if for no other reason that I'd wonder why Microsoft would put in arbitrary requirements, and this would clarify that. That being said, an example would help, and you could certainly look at it faster.